torch/_refs/__init__.py

```diff
@@ -1658,49 +1658,135 @@ def item(a: TensorLikeType) -> NumberType:
+    if isinstance(arg0, TensorLike):
+        # overload to `to.other`
+        device = arg0.device
+        dtype = arg0.dtype
+        layout = arg0.layout
+        # TODO: is_pinned is not currently supported in refs or fake_tensor
+        # https://github.com/pytorch/pytorch/issues/84925
+        # pin_memory = arg0.is_pinned()
+        # load positional arg_listuments:
+        if arg_list:
+            assert non_blocking is None
+            non_blocking, *arg_list = arg_list
+        if arg_list:
+            assert copy is None
+            copy, *arg_list = arg_list
+        if arg_list:
+            assert memory_format is None
+            memory_format, *arg_list = arg_list
+    elif isinstance(arg0, torch.dtype):
+        # overload to `to.dtype`
+        dtype = arg0
+        if arg_list:
+            assert non_blocking is None
+            non_blocking, *arg_list = arg_list
+        if arg_list:
+            assert copy is None
+            copy, *arg_list = arg_list
+        if arg_list:
+            assert memory_format is None
+            memory_format, *arg_list = arg_list
+    elif isinstance(arg0, torch.device):
+        # overload to `to.device`
+        device = arg0
+        if arg_list:
+            assert dtype is None
+            dtype, *arg_list = arg_list
+        if arg_list:
+            assert non_blocking is None
+            non_blocking, *arg_list = arg_list
+        if arg_list:
+            assert copy is None
+            copy, *arg_list = arg_list
+        if arg_list:
+            assert memory_format is None
+            memory_format, *arg_list = arg_list
+    else:
+        # overload to `to.dtype_layout` takes no positional arg_lists
+        assert arg0 is None
+    # assert that we have processed all positional argument
+    assert not arg_list
+    return device, dtype, non_blocking, copy, memory_format, layout, pin_memory
+
+
+def to(a: TensorLikeType, *args, **kwargs) -> TensorLikeType:
+    device, dtype, non_blocking, copy, memory_format, layout, pin_memory = _to_dispatch(
+        a, *args, **kwargs
+    )
+
+    if _to_will_alias(a, device, dtype, copy, memory_format, layout):
         return a
+    assert pin_memory is None
+    # short-circuit to `prims.convert_element_type` when `to` is just a dtype change
     if (
-        (copy is True or dtype != a.dtype)
-        and memory_format == torch.preserve_format
-        and non_blocking is False
+        (copy or dtype != a.dtype)
+        and (memory_format is None or memory_format == torch.preserve_format)
+        and (not non_blocking)
+        and (device is None or device == a.device)
+        and (layout is None or layout == a.layout)
+        # TODO: is_pinned is not currently supported in refs or fake_tensor
+        # https://github.com/pytorch/pytorch/issues/84925
+        # and (pin_memory is None or pin_memory == a.is_pinned())
     ):
         return prims.convert_element_type(a, dtype)
+    memory_format = (
```

Comment:
Seems like this should go earlier, before `memory_format` is queried as part of the test for whether this can short-circuit to `prims.convert_element_type`? Maybe it makes more sense to have `memory_format` set properly from `_to_dispatch`?